src/main/java/ch/njol/skript/conditions/CondIsNamed.java

```diff
@@ -0,0 +1,93 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.SkriptConfig;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.conditions.base.PropertyCondition.PropertyType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.ExprName;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is Named")
+@Description("Checks whether or not a item, block, slot, inventory, player or entity is named or has a given name")
+@Examples({
+	"on right click with a cake:",
+	"\tif event-item is named \"&bBirthday Cake\"",
+	"\t\tsend \"Happy Birthday, %player's name%!\"",
+	"",
+	"on damage of a player:",
+	"\tattacker's tool is named",
+	"\tcancel event",
+	"",
+	"on inventory click:",
+	"\tevent-inventory is named \"Example Inventory\":",
+	"\t\tcancel event",
+	""
+})
+@Since("INSERT VERSION")
+public class CondIsNamed extends Condition {
+
+	@Nullable
+	private Expression<String> name;
+	private Expression<Object> objects;
+	private boolean caseSensitive = false;
+	private static final ExprName exprName = new ExprName();
+
+	static {
+		PropertyCondition.register(CondIsNamed.class, PropertyType.BE, "named [%-string%]", "offlineplayers/entities/blocks/itemtypes/inventories/slots/worlds");
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		objects = (Expression<Object>) exprs[0];
+		name = (Expression<String>) exprs[1];
+		caseSensitive = SkriptConfig.caseSensitive.value();
+		parseResult.mark = 1;
+		exprName.init(exprs, 0, isDelayed, parseResult);
+		setNegated(matchedPattern == 1);
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		String name = this.name != null ? this.name.getSingle(event) : null;
+		return objects.check(event, object -> {
+			String value = exprName.convert(object);
+			return name != null ? StringUtils.equals(name, value, caseSensitive) : value != null;
```

Comment:
Rather than doing this name != null, I think right after defining variable name, if it's null it should return false no matter what, because Skript cannot check if the name is something if the provided string is null, so return false regardless of negation.

Reply:
I'm not sure I follow what you mean on this part. Since name can't ever be a Boolean it's either null or String. `name` refers to the `named %-string%` argument. I should probably make it a better name for those vs. `value` and `name`

Reply:
between line 80 and 81 should be a return if name is null

Reply:
ah alright yeah that's do able for me give me a second or 10

Reply:
> between line 80 and 81 should be a return false if name is null

actually name being null was designed for that as I wanted to allow `player's tool is named`. so wouldn't that just remove that entire functionality?

Reply:
Went with a method I showed above which includes 1 extra check to remove any possible null issues with the equals. If this wasn't what you wanted let me know